daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -413,9 +413,11 @@ object Speedy {
 
           // start evaluating the arguments
           val newArgsLimit = Math.min(missing, newArgs.length)
-          for (i <- 1 until newArgsLimit) {
+          var i = 1
+          while (i < newArgsLimit) {
```

Comment:
Yourkit showed a lot of RichInt conversions, so I converted this iterator to `while {}` style

Reply:
Same story here. I would like to have real benchmarks before changing critical piece of code like that. 
cc @dajmaki

Reply:
If @dajmaki  does not mind, I am fine with this change,

Reply:
I would not hurt to run `com.digitalasset.daml.lf.engine.LargeTransactionTest` before and after the change. There is also `com.digitalasset.platform.sandbox.perf.LargeTransactionBench` which is supposed to be part of perf test harness.

Reply:
Again this was found after running our load tests and profiling the code with YourKit.
Using ScalaMeter to show some numbers:
1. Quick but ugly code:
```      val length = arr
      var i = 0
      while (i < length) {
        i += 1
      }
```
results : Parameters(size -> 15000000): 5.36E-4 ms

2. Slow but pretty code:
```      for (_ <- 0 until upper) {}```
results: Parameters(size -> 15000000): 3.671207 ms

Again, that would not be an issue if this part of code was not so frequently called.

Reply:
FYI `scala.collection` uses similar optimizations. :+1: for `var` and `while-loop`.